sympy/integrals/rubi/symbol.py

```diff
@@ -0,0 +1,20 @@
+from sympy.external import import_module
+matchpy = import_module("matchpy")
+
+if matchpy:
+    Symbol = matchpy.Symbol
+else:
+    class Symbol(object):
+        def __init__(self, name, variable_name=None):
+            self.name = name
+            self.head = self
+        @staticmethod
+        def dot(x):
+            pass
+
+class VariableSymbol(Symbol):
+    pass
+
+class Integer(Symbol):
```

Comment:
There could be a potential name conflict with SymPy's Integer class. I generally suggest not to give the same name to two different classes in SymPy.